standards/API_STANDARDS.md

```diff
@@ -11,29 +11,40 @@ API standards
   * Search/list endpoints should contain consistent fields:
     * Response: `count`, `limit`, `offset`, `total_count`
 
-	  When pagination is required, the fields work as:
+	  When pagination is required, it is also required the data be in a sorted order before it can be paginated. The fields work as:
 
-	  `limit` - max number of items we're returning in this response (e.g. 20, or 50, rather than all we use atm)
+	  `limit` - max number of items we're returning in this response.
+	  * If value is not set, a configurable default limit should be used, unless otherwise stated this should be set to a value of `20`.
+	  * The limit value can be set to `0` to return `0` items, so an API user can obtain the metadata for the list endpoint.
```

Comment:
The current implementation assumes that 0 means 'return all items'. It's the same assumption that mongoDB does, for example: if we pass limit=0 to the following mongoDB query
```
	iter := q.Sort("option").Skip(offset).Limit(limit).Iter()
```
Then it returns all the items.

This change would have an advantage: if we need to return 0 items, we actually only need to perform a Count query, which has better performance. But maybe its disadvantage would be that it requires extra code and users will think that 0 means everything?
Do we want to allow an extra value to return everything?

Reply:
I've just seen the comment about setting a maximum possible value of 500 or 1000 so that callers can't cause responses that would be too large or have a performance impact.
In that case, maybe there is no point in allowing a value for 'return everything'?

Reply:
Do users think 0 means everything? Should we be allowing users to get everything? <- This doesn't seem sensible